src/query/interpreter.cpp

```diff
@@ -1909,6 +1918,119 @@ PreparedQuery PrepareSettingQuery(ParsedQuery parsed_query, const bool in_explic
+          std::string status{"Transaction not found"};
+          interpreter_context->interpreters.WithLock(
+              [db_accessor, &transaction_id, &status, username, isAdmin](const auto &interpreters) {
+                for (Interpreter *interpreter : interpreters) {
+                  std::string intr_trans_str = std::to_string(interpreter->GetTransactionId());
+                  if (interpreter->HasActiveTransaction() && transaction_id == intr_trans_str) {
+                    if (interpreter->username_ == username || isAdmin) {
+                      // if (interpreter->in_explicit_transaction_) {
+                      //   interpreter->expect_rollback_ = true;
+                      // } else {
+                      interpreter->AbortTransactionByUser();
+                      interpreter->Abort();
+                      // }
+                      status = "Transaction killed";
+                    } else {
+                      status = "Not enough rights to kill the transaction";
+                    }
+                    break;
+                  }
+                }
+              });
+          results.push_back({TypedValue(transaction_id), TypedValue(status)});
+        }
+        return results;
+      };
+      break;
+    }
+  }
+
+  return callback;
+}
+
+PreparedQuery PrepareTransactionQueueQuery(ParsedQuery parsed_query, const std::optional<std::string> &username,
+                                           const uint64_t current_transaction_id, const bool in_explicit_transaction,
+                                           InterpreterContext *interpreter_context, DbAccessor *dba) {
+  if (in_explicit_transaction) {
+    // throw SettingConfigInMulticommandTxException{};
+  }
+
+  spdlog::info("Number of interpreters in prepare transactions: {}", interpreter_context->interpreters->size());
+
+  auto *transaction_queue_query = utils::Downcast<TransactionQueueQuery>(parsed_query.query);
+  MG_ASSERT(transaction_queue_query);
+  auto callback = HandleTransactionQueueQuery(transaction_queue_query, username, current_transaction_id,
+                                              parsed_query.parameters, interpreter_context, dba);
+
+  return PreparedQuery{std::move(callback.header), std::move(parsed_query.required_privileges),
+                       [callback_fn = std::move(callback.fn), pull_plan = std::shared_ptr<PullPlanVector>{nullptr}](
+                           AnyStream *stream, std::optional<int> n) mutable -> std::optional<QueryHandlerResult> {
+                         if (UNLIKELY(!pull_plan)) {
+                           pull_plan = std::make_shared<PullPlanVector>(callback_fn());
+                         }
+
+                         if (pull_plan->Pull(stream, n)) {
+                           return QueryHandlerResult::COMMIT;
+                         }
+                         return std::nullopt;
+                       },
+                       RWType::NONE};
```

Comment:
please review this part with Benjo or someone more experienced, I'm not sure what's a PullPlanVector

Reply:
or UNLIKELY

Reply:
yes